src/npyffi/types.rs

```diff
@@ -201,3 +201,70 @@ pub struct npy_stride_sort_item {
     pub perm: npy_intp,
     pub stride: npy_intp,
 }
+
+#[repr(u8)]
+#[derive(Debug, Clone, Copy)]
+pub enum NPY_TYPECHAR {
+    NPY_BOOLLTR = b'?',
+    NPY_BYTELTR = b'b',
+    NPY_UBYTELTR = b'B',
+    NPY_SHORTLTR = b'h',
+    NPY_USHORTLTR = b'H',
+    NPY_INTLTR = b'i',
+    NPY_UINTLTR = b'I',
+    NPY_LONGLTR = b'l',
+    NPY_ULONGLTR = b'L',
+    NPY_LONGLONGLTR = b'q',
+    NPY_ULONGLONGLTR = b'Q',
+    NPY_HALFLTR = b'e',
+    NPY_FLOATLTR = b'f',
+    NPY_DOUBLELTR = b'd',
+    NPY_LONGDOUBLELTR = b'g',
+    NPY_CFLOATLTR = b'F',
+    NPY_CDOUBLELTR = b'D',
+    NPY_CLONGDOUBLELTR = b'G',
+    NPY_OBJECTLTR = b'O',
+    NPY_STRINGLTR = b'S',
+    NPY_STRINGLTR2 = b'a',
+    NPY_UNICODELTR = b'U',
+    NPY_VOIDLTR = b'V',
+    NPY_DATETIMELTR = b'M',
+    NPY_TIMEDELTALTR = b'm',
+    NPY_CHARLTR = b'c',
+    NPY_INTPLTR = b'p',
+    NPY_UINTPLTR = b'P',
+}
+
+#[repr(u8)]
+#[derive(Debug, Clone, Copy)]
+// NPY_TYPEKINDCHAR doesn't exist in the header, but these enum values are not
+// related to NPY_TYPECHAR although being stuffed into it (type kinds, not type codes)
+pub enum NPY_TYPEKINDCHAR {
```

Comment:
In the original header these are part of `NPY_TYPECHAR`, but they have repeated values + a comment that they don't belong there.

Reply:
Is there a strong reason why we should differ from the C header?

Reply:
Just one reason really - Rust enums can't have repeated values... (all 5 of these values already occur in `NPY_TYPECHAR')

Reply:
(perhaps the comment should reflect that as well)

Reply:
(Updated the comment preceding this enum, hope it's more clear now)